exonum/src/runtime/mod.rs

```diff
@@ -375,11 +414,15 @@ pub enum Caller {
         /// Public key of the user who signed this transaction.
         author: PublicKey,
     },
+
     /// Method is invoked during the method execution of a different service.
     Service {
         /// Identifier of the service instance which invoked this method.
         instance_id: InstanceId,
     },
+
+    /// Method is invoked by the `before_commit` hook.
+    BeforeCommit,
```

Comment:
It could make sense to generalize this to `Blockchain` (as proposed by @alekseysidorov), or to split into `BeforeCommit`, `GenesisBlock`, etc.

Reply:
imo: If we split this options, it will sounds like a provocation to consider these cases as completely different.